src/components/ButtonIcon/styled/button.js

```diff
@@ -71,60 +83,70 @@ const StyledButton = styled.button`
         }
     }
 
-    ${props =>
-        props.variant === 'brand' &&
-        `
-            background-color: ${COLOR_BRAND};
-            border: 1px solid ${COLOR_BRAND};
-            color: ${COLOR_WHITE};
+    ${props => {
+        const brandMainContrastText = props.getContrastText(props.brandMainColor);
+        return (
+            props.variant === 'brand' &&
+            `
+                background-color: ${props.brandMainColor};
+                border: 1px solid ${props.brandMainColor};
+                color: ${brandMainContrastText};
+                
+                &:link,
+                &:visited,
+                &:active {
+                    color: ${brandMainContrastText};
+                }
             
-            &:link,
-            &:visited,
-            &:active {
-                color: ${COLOR_WHITE};
-            }
-        
-            &:hover,
-            &:focus,
-            &:active {
-                background-color: ${COLOR_BRAND_ACTIVE};
-                border: 1px solid ${COLOR_BRAND_ACTIVE};
-                color: ${COLOR_WHITE};
-            }
-        
-            &[disabled] {
-                background-color: ${COLOR_GRAY_1};
-                border: 1px solid ${COLOR_GRAY_1};
-                color: ${COLOR_GRAY_2};
-            }
-        `};
-    ${props =>
-        props.variant === 'success' &&
-        `
-            background-color: ${COLOR_SUCCESS};
-            border: 1px solid ${COLOR_SUCCESS};
-            color: ${COLOR_WHITE};
+                &:hover,
+                &:focus,
+                &:active {
+                    background-color: ${props.brandDarkColor};
+                    border: 1px solid ${props.brandDarkColor};
+                    color: ${brandMainContrastText};
```

Comment:
This contrast text needs to be computed from brandDarkColor, not from brandMainColor.